core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -698,9 +704,13 @@ private[spark] class TaskSetManager(
         totalResultSize -= resultSizeAcc.get.asInstanceOf[LongAccumulator].value
       }
 
+      val killedReason = if (killedByOtherAttempt.contains(tid)) {
+        TaskKilled("Finish but did not commit due to another attempt succeeded")
+      } else {
+        TaskKilled("Finish but did not commit due to task in another stage retry succeeded")
```

Comment:
is it really worth to add `killedByOtherStageRetries` just for a slightly better error message?

Reply:
`killedByOtherStageRetries` is also used in `executorLost` to avoid `Resubmitted`.